autopilot/hostinfo.go

```diff
@@ -65,66 +65,72 @@ func (c *contractor) HostInfo(ctx context.Context, hostKey types.PublicKey) (api
 	}, nil
 }
 
-func (c *contractor) HostInfos(ctx context.Context, filterMode, addressContains string, keyIn []types.PublicKey, offset, limit int) ([]api.HostHandlerGET, error) {
-	cfg := c.ap.Config()
-	if cfg.Contracts.Allowance.IsZero() {
-		return nil, fmt.Errorf("can not score hosts because contracts allowance is zero")
-	}
-	if cfg.Contracts.Amount == 0 {
-		return nil, fmt.Errorf("can not score hosts because contracts amount is zero")
-	}
-	if cfg.Contracts.Period == 0 {
-		return nil, fmt.Errorf("can not score hosts because contract period is zero")
-	}
+func (c *contractor) HostInfos(ctx context.Context, filterMode, usabilityMode, addressContains string, keyIn []types.PublicKey, offset, limit int) ([]api.HostHandlerGET, error) {
+	c.mu.Lock()
+	hostInfo := c.cachedHostInfo
+	c.mu.Unlock()
 
-	hosts, err := c.ap.bus.SearchHosts(ctx, filterMode, addressContains, keyIn, offset, limit)
-	if err != nil {
-		return nil, fmt.Errorf("failed to fetch requested hosts from bus: %w", err)
-	}
-	gs, err := c.ap.bus.GougingSettings(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to fetch gouging settings from bus: %w", err)
-	}
-	rs, err := c.ap.bus.RedundancySettings(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to fetch redundancy settings from bus: %w", err)
-	}
-	cs, err := c.ap.bus.ConsensusState(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to fetch consensus state from bus: %w", err)
-	}
-	fee, err := c.ap.bus.RecommendedFee(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to fetch recommended fee from bus: %w", err)
+	// declare helper to decide whether to keep a host.
+	if usabilityMode != api.UsabilityFilterModeUsable && usabilityMode != api.UsabilityFilterModeUnusable && usabilityMode != api.UsabilityFilterModeAll {
```

Comment:
`nit` extract into `isValidUsabilityMode` and perform this check before grabbing the cache